packages/cli/lib/LocalDevManager.js

```diff
@@ -143,36 +160,53 @@ class LocalDevManager {
   }
 
   logUploadWarning(reason) {
-    const currentDefaultAccount = getConfigDefaultAccount();
-    const defaultAccountId = getAccountId(currentDefaultAccount);
+    if (!this.uploadWarnings[reason]) {
+      const currentDefaultAccount = getConfigDefaultAccount();
+      const defaultAccountId = getAccountId(currentDefaultAccount);
 
-    logger.log();
-    logger.warn(i18n(`${i18nKey}.uploadWarning.header`, { reason }));
-    logger.log(
-      i18n(`${i18nKey}.uploadWarning.stopDev`, {
-        command: uiCommandReference('hs project dev'),
-      })
-    );
-    if (this.targetAccountId !== defaultAccountId) {
+      logger.log();
+      logger.warn(i18n(`${i18nKey}.uploadWarning.header`, { reason }));
       logger.log(
-        i18n(`${i18nKey}.uploadWarning.runUploadWithAccount`, {
-          command: uiCommandReference(
-            `hs project upload --account=${this.targetAccountId}`
-          ),
+        i18n(`${i18nKey}.uploadWarning.stopDev`, {
+          command: uiCommandReference('hs project dev'),
         })
       );
-    } else {
+      if (this.targetAccountId !== defaultAccountId) {
+        logger.log(
+          i18n(`${i18nKey}.uploadWarning.runUploadWithAccount`, {
+            command: uiCommandReference(
+              `hs project upload --account=${this.targetAccountId}`
+            ),
+          })
+        );
+      } else {
+        logger.log(
+          i18n(`${i18nKey}.uploadWarning.runUpload`, {
+            command: uiCommandReference('hs project upload'),
+          })
+        );
+      }
       logger.log(
-        i18n(`${i18nKey}.uploadWarning.runUpload`, {
-          command: uiCommandReference('hs project upload'),
+        i18n(`${i18nKey}.uploadWarning.restartDev`, {
+          command: uiCommandReference('hs project dev'),
         })
       );
+
+      this.mostRecentUploadWarning = reason;
+      this.uploadWarnings[reason] = true;
     }
-    logger.log(
-      i18n(`${i18nKey}.uploadWarning.restartDev`, {
-        command: uiCommandReference('hs project dev'),
-      })
-    );
+  }
+
+  monitorConsoleOutput() {
```

Comment:
This allows us to intercept any `console.log` calls to help prevent us from spamming upload warnings to the console. It pairs with the logic in `logUploadWarning`. The behavior is like this:

If the last thing logged to the console was an upload warning...
- Prevent any upload warning that has the exact same `reason` from logging to the console again.
- Allow any upload warning that has a different `reason` to log to the console
- Allow any other console logs that are not upload warnings.

If the last thing logged to the console was not an upload warning...
- Allow all console logs, including upload warnings